packages/schema/compose/src/resolve.ts

```diff
@@ -1028,41 +1042,41 @@ async function resolveLocalImports(
   }
 }
 
-export function resolveEnvTypes(typeInfo: TypeInfo, mutation: boolean): void {
-  const genericEnvType = typeInfo.objectTypes.find(
-    (type) => type.type === "Env"
-  );
-  if (!genericEnvType) {
+export function resolveEnvTypes(
+  typeInfo: TypeInfo,
+  schemaKind: SchemaKind
+): void {
+  const sharedEnvDef = typeInfo.objectTypes.find((type) => type.type === "Env");
+  if (!sharedEnvDef) {
     return;
   }
 
-  const specificEnvType = mutation
+  const isMutationEnv = schemaKind === "mutation" || schemaKind === "plugin";
+
+  const moduleEnvDef = isMutationEnv
     ? typeInfo.envTypes.mutation
     : typeInfo.envTypes.query;
 
-  if (!specificEnvType.sanitized) {
-    specificEnvType.sanitized = createObjectDefinition({
-      type: mutation ? envTypes.MutationEnv : envTypes.QueryEnv,
+  if (!moduleEnvDef.sanitized) {
+    moduleEnvDef.sanitized = createObjectDefinition({
+      type: isMutationEnv ? envTypes.MutationEnv : envTypes.QueryEnv,
     });
   }
 
   typeInfo.objectTypes = typeInfo.objectTypes.filter((type) => {
-    return type.type !== genericEnvType.type;
+    return type.type !== sharedEnvDef.type;
   });
 
-  checkDuplicateEnvProperties(
-    specificEnvType.sanitized,
-    genericEnvType.properties
-  );
-  specificEnvType.sanitized.properties.push(...genericEnvType.properties);
+  checkDuplicateEnvProperties(moduleEnvDef.sanitized, sharedEnvDef.properties);
+  moduleEnvDef.sanitized.properties.push(...sharedEnvDef.properties);
 }
 
 export function checkDuplicateEnvProperties(
   envType: ObjectDefinition,
-  genericEnvProperties: AnyDefinition[]
+  envProperties: AnyDefinition[]
 ): void {
   const genericEnvPropertiesSet = new Set(
```

Comment:
Did you missed this refactor? `genericEnvPropertiesSet` -> `envPropertiesSet `

Reply:
Yes, I felt it was a bit more confusing than it had to be using the word `generic`. This function is simply checking to see if an array of properties exist on a type. The type in this case being `envType`, and the properties being `envProperties`. It helps it be easier to read and understand IMO.